### PR TITLE
feat(plugin): trivy-aws as a plugin for Trivy

### PR DESCRIPTION
**Pull request by simar7**
## Notes
1. The current `aws` subcommand will cause conflict if we name the plugin the same. So I've named it `aws-scan` for now.
2. We need to deprecate the current `aws` subcommand within trivy. This will still continue to work until we remove it however. Will done in a separate PR. 

### Install

```shell

$ trivy plugin install ./trivy-aws.tar.gz
2024-05-18T00:06:06-06:00       INFO    Installing the plugin...        url="./trivy-aws.tar.gz"
2024-05-18T00:06:08-06:00       INFO    Loading the plugin metadata...

```

### Run

```shell
$ AWS_REGION=<region> AWS_ACCOUNT_ID=<id> trivy aws-scan
2024/05/18 00:06:23 INFO Need to update the built-in policies
2024/05/18 00:06:23 INFO Downloading the built-in policies...

<results>
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,16 +2,24 @@
 test:
 	go test -race ./...
 
+.PHONY: build
+build:
+	go build -o trivy-aws ./cmd/trivy-aws/main.go
```
Comment (nikpivkin) [inline Makefile:7]: You can reduce the size of the executable binary:
```suggestion
	go build -ldflags "-s -w" -o trivy-aws ./cmd/trivy-aws/main.go
```
```
trivy-aws: 273M -> 187M
trivy-aws.tar.gz: 112M -> 46M
```

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/iac/state"
+)
+
+type Cache struct {
```
Comment (nikpivkin) [inline pkg/cache/cache.go:15]: Is this cache specific to AWS only, or can it be used for other cloud providers in the future?

File: pkg/commands/app.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		MisconfFlagGroup: flag.NewMisconfFlagGroup(),
+		RegoFlagGroup:    flag.NewRegoFlagGroup(),
+		ReportFlagGroup:  reportFlagGroup,
+	}
+
+	services := scanner.AllSupportedServices()
+	sort.Strings(services)
+
+	cmd := &cobra.Command{
+		Use:     "aws [flags]",
```
Comment (nikpivkin) [inline pkg/commands/app.go:41]: Should be `aws-scan'. The long description of the command should also be updated.

Comment (nikpivkin) [inline pkg/commands/app.go:41]:
```suggestion
		Use:     "aws-scan [flags]",
```

The long description of the command should also be updated.

File: pkg/cache/cache.go (hunk 2 above)
Comment (simar7) [reply pkg/cache/cache.go:15]: For now just AWS. If we ever decide to add other cloud providers, we can reconsider it.

File: pkg/commands/app.go (hunk 3 above)
Comment (simar7) [reply pkg/commands/app.go]: I've updated it for now but I think we should finalize the plugin name first. I have `aws-scan` for now as I couldn't come up with anything better.

So `trivy aws-scan <flags>`.

File: Makefile (hunk 1 above)
Comment (simar7) [reply Makefile]: Nice, updated.

**Comment by nikpivkin** [comment]
@simar7 Updated the docker version in the `go-mock-aws` package. https://github.com/aquasecurity/go-mock-aws/pull/5

**Comment by knqyf263** [comment]
I think we want to add this plugin to [the index](https://aquasecurity.github.io/trivy-plugin-index/) so people can install this plugin by `trivy plugin install aws`.

**Comment by simar7** [comment]
> I think we want to add this plugin to [the index](https://aquasecurity.github.io/trivy-plugin-index/) so people can install this plugin by `trivy plugin install aws`.

Yes I was going to do just that 👍🏼

**Comment by knqyf263** [comment]
@simar7 Added!
CleanShot 2024-05-31 at 09 00 50

**Comment by nikpivkin** [comment]
Some packages such as `cache`, `commands` and `scanner` are duplicated in `pkg` and `pkg/cloud/aws`.

```sh
❯ ls pkg/cloud/aws
cache    commands scanner
❯ ls pkg/
cache       cloud       commands    concurrency errs        progress    scanner     types
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (nikpivkin) [inline .github/workflows/release.yml:10]: Should we build the plugin for the same platforms as Trivy to maintain backwards compatibility?

Comment (simar7) [reply .github/workflows/release.yml:10]: I think we can add cross platform support later on if asked for. Feels like feature creep for this PR.

**Review by nikpivkin — APPROVED** [review]

**Pull request merged**